plugin/xule/__init__.py

```diff
@@ -775,52 +775,49 @@ def xuleCmdXbrlLoaded(cntlr, options, modelXbrl, *args, **kwargs):
         runXule(cntlr, options, modelXbrl)
 
 def runXule(cntlr, options, modelXbrl, rule_set_map=_xule_rule_set_map_name):
-        try:
-            if getattr(options, "xule_multi", True) and \
-                getattr(cntlr, "rule_set", None) is not None:
-                rule_set = getattr(cntlr, "rule_set")
-            else:
-                if getattr(options, 'xule_rule_set', None) is not None:
-                    rule_set_location = options.xule_rule_set
-                else:
-                    # Determine the rule set from the model.
-                    rule_set_location = xu.determine_rule_set(modelXbrl, cntlr, rule_set_map)
-                    modelXbrl.log('INFO', 'info', 'Using ruleset {}'.format(rule_set_location))
-                    if rule_set_location is None:
-                        # The rule set could not be determined.
-                        modelXbrl.log('ERROR', 'xule', "Cannot determine which rule set to use for the filing. Check the rule set map at '{}'.".format(xu.get_rule_set_map_file_name(cntlr, rule_set_map)))
-
-                rule_set = xr.XuleRuleSet(cntlr)
-                rule_set.open(rule_set_location, open_packages=not getattr(options, 'xule_bypass_packages', False))
-        except xr.XuleRuleSetError:
-            raise
-
-        if getattr(options, "xule_multi", False):
-            xm.start_process(rule_set,
-                         modelXbrl,
-                         cntlr,
-                         options
-                         )
+    if getattr(options, "xule_multi", True) and getattr(cntlr, "rule_set", None) is not None:
```

Comment:
Should this default to `False` instead of `True`?

Reply:
Im not positive, but I would assume that XULE would always want to be multi threaded even if the xule_multi variable isnt set.

Reply:
This should be switched back to ``False`` for the default for the xule_muli option so it reads:

``if getattr(options, "xule_multi", False) and getattr(cntlr, "rule_set", None) is not None:``

The --xule-multi option is only used when running xule in a server. It is not multi threading, but multi processing. For xule, this requires running xule in the xule server mode with the Xule_Multi module. The Xule_Multi module is not part of the DQC distribution. It signals the server to create multiple copies of Arelle in different processes and do some preloading of calculated constants. The server than can split the work between the processes. Its basically a poor man version of multi threading. But it requires some updates to Arelle itself, so it doesn't run with the official distribution of Arelle.

In normal running, the rule set is not supplied, so this IF statement will evaluate to False and execute the ELSE clauses. So it probably doesn't affect the normal run. But leaving this defaulting to True here could cause some unexpected issues. when running with a supplied rule set.

@davidtauriello will switch this back to False after PR #457 is merged.